graphql/provider.go

```diff
@@ -24,6 +31,27 @@ func Provider() *schema.Provider {
 				Optional: true,
 				ForceNew: true,
 			},
+			"login_query": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"login_query_variables": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				ForceNew: true,
+			},
+			"authenticated_headers": {
+				Type: schema.TypeMap,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+				ForceNew: true,
```

Comment:
Ive never really thought about this, but what is the behavior of `force new` when its used on a provider column? I assume it just re-inits the provider in state, but want to make sure it wont try to recreate all of the corresponding resources. Can you do a little manual test by changing any of these values between runs and ensure the provider wont blow everything away? I'm unsure at this point if we need to set force_new on these, since as we saw in your other manual test, the login_query appears to run fresh on every lifecycle, so no need to blow away the provider in state.

Reply:
I do not believe it does anything at the provider level. I'll check.

Reply:
So, using `ForceNew` on a field is only meaningful in the case of a resource as [it means](https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#forcenew):
> `ForceNew` indicates that any change in this field requires the _resource to be destroyed and recreated_

In which case the resource's [`DeleteContext`](https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.10.1/helper/schema/resource.go#L146) and [`CreateContext`](https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.10.1/helper/schema/resource.go#L143) functions [will be invoked](https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.10.1/helper/schema/resource.go#L436).

There's no equivalent handling for it in the provider schema code: https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.10.1/helper/schema/provider.go
The schema fields will only be used in during provider configuration: https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.10.1/helper/schema/provider.go#L261

Note that it is also meaningless for a data source which is a special case of resource with only a read function, but no create, update and delete functions.

There is even a check in [tfproviderlint](https://github.com/bflad/tfproviderlint) to report when it is set on a datasource's field: https://github.com/bflad/tfproviderlint/tree/v0.27.1/passes/S024

Should I submit another PR removing the useless `ForceNew` on all provider and data source fields?

Reply:
No need to submit another PR unless you want to :). I would be ok if you removed the ForceNew stuff on the data source in this PR as that would be easier for you. The impact of that will be 0 so Im not worried about it

Reply:
Done.